libwacom/libwacom-database.c

```diff
@@ -295,32 +412,69 @@ libwacom_parse_stylus_keyfile(WacomDeviceDatabase *db, const char *path)
 
 	groups = g_key_file_get_groups (keyfile, NULL);
 	for (i = 0; groups[i]; i++) {
-		WacomStylus *stylus;
+		WacomStylus *stylus = NULL, *aliased = NULL;
 		WacomStylusId id;
 		GError *error = NULL;
 		char *eraser_type, *type;
 		char **string_list;
+		int aliased_int;
 
 		if (!parse_stylus_id(groups[i], &id)) {
 			g_warning ("Failed to parse stylus ID '%s', ignoring entry", groups[i]);
 			continue;
 		}
+
 		stylus = g_new0 (WacomStylus, 1);
 		stylus->refcnt = 1;
 		stylus->id = id;
-		stylus->name = g_key_file_get_string(keyfile, groups[i], "Name", NULL);
-		stylus->group = g_key_file_get_string(keyfile, groups[i], "Group", NULL);
+
+		char *aliasstr = g_key_file_get_string(keyfile, groups[i], "AliasOf", NULL);
```

Comment:
Note that `aliasstr` leaks in every `continue` case below (except the `!aliasstr` one)